.github/workflows/publish.yaml

```diff
@@ -0,0 +1,14 @@
+# A CI configuration to auto-publish pub packages.
+
+name: Publish
+
+on:
+  pull_request:
+    branches: [ master ]
```

Comment:
why does publish run on a PR?

Reply:
It validates the info - so you know that it's something you _could_ publish - and it writes a comment to the PR telling you the git tag to use if everything looks good for publishing.

So, validation on PRs, and, publishing on well-formed commits.

Reply:
Clever clever...